### PR TITLE
refactor: moves config details out of quick start

### PR DESCRIPTION
**Pull request by mirestrepo**

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -5,27 +5,52 @@ title: Configuration
 
 ## Environment Variables
 
-Honeycomb uses environment variables during build and run time to control the different configurations. Below we list the  environment variables used by the app and indicate whether they are mandatory, optional and needed during build (`npm build`) or run time (using the application executable or `npm run dev`)
+Honeycomb uses environment variables during **build** and **run** time to control the different configurations. Below we list the  environment variables used by the app and indicate (via badges and text), whether they are mandatory, optional and needed during build (`npm build`) or run time (using the application executable or `npm run dev`). We also separet them via the enviroment (Desktop or Online) where get used by.
```
Comment (RobertGemmaJr) [inline docs/configuration.md]:
```suggestion
Honeycomb uses environment variables during **build** and **run** time to control the different configurations. Below we list the  environment variables used by the app and indicate (via badges and text), whether they are mandatory, optional and needed during build (`npm build`) or run time (using the application executable or `npm run dev`). We also separate them based on the environment (Desktop or Online) where honeycomb will be run.
```

**Review by gtdang — COMMENTED** [review]
> Below we list the environment variables used by the app and indicate (via badges and text), whether they are mandatory, optional and needed during build (`npm build`) or run time (using the application executable or `npm run dev`).

I'm a bit confused by this sentence and the badges used in this guide. Perhaps splitting this sentence into two, 1) explain the optional/mandatory and 2) runtime/buildtime would be beneficial.
* Optional vs Mandatory - Should explain that the `optional` badge indicates optional and `default` + `default value` badge indicates mandatory?  

*Suggested edit*
Below we list the environment variables used by the app and indicate their properties via badges and text. Optional variables are labeled with the optional badge and mandatory variables are labeled with the default false badge with their default value specified. The buildtime badge indicates the variable is used during the build/compilation phase (`npm build`) while the runtime badge indicates the variable is used during the executable run (`nom run dev`).

File: docs/quick_start.md [hunk 2]
```diff
@@ -7,9 +7,15 @@ To start a new task locally, in development mode follow these steps:
 
 ### 1. Start your new task from our template repository
 
-The simplest way to get started is creating a new repository using Honeycomb as a template.
+The simplest way to get started is creating a new repository using Honeycomb as a template. This option will allow you to generate a new repository with the same directory structure and files as an existing repository.
```
Comment (RobertGemmaJr) [inline docs/quick_start.md]:
```suggestion
The simplest way to get started is creating a new repository using Honeycomb as a template. This option will generate a new repository with the same directory structure and files as the existing repository.
```

File: docs/quick_start.md [hunk 3]
```diff
@@ -111,28 +117,15 @@ npm run dev
...
-```
-npm run dev:clinic
-```
-```
-npm run dev:clinic:video
-```
-```
-npm run dev:firebase
-```
+HoneyComb tasks can be configured to run as a webapp in Firebase, as desktop application via electron, to receive port signals from EEG, cameras, foot pedals and more. To learn more about how ro configure your task for these different scenarios, see the [Connfiguration Section](/docs/configuration).
```
Comment (RobertGemmaJr) [inline docs/quick_start.md]: Also I think it's "Honeycomb" not "HoneyComb"? Not 100% sure
```suggestion
HoneyComb tasks can be configured to run as a web app in Firebase, as desktop application via electron, to receive port signals from EEG, cameras, foot pedals and more. To learn more about how to configure your task for these different scenarios, see the [Configuration Section](/docs/configuration).
```

File: docs/quick_start.md [hunk 4]
```diff
@@ -111,28 +117,15 @@ npm run dev
...
 
 ### 7. Check out the data
 
-The data is saved throughout the task to the users's app directory.  This is logged at the beginning of the task wherever you ran `npm run dev`. It is also stored in a folder that is generated by the app, which should be found on the desktop.
+The data is saved throughout the task. When running as a Deskto App, data is saved to the users's app directory. When running as Web App in Firebase, data is stored in the FireStore. 
+
+For a Desktop App, the location of the  is logged at the beginning of the task wherever you ran `npm run dev`. Data is also stored in a folder that is generated by the app, which should be found on the desktop.
```
Comment (RobertGemmaJr) [inline docs/quick_start.md]:
```suggestion
The data is saved throughout the task. When running as a Desktop App, data is saved to the user's app directory. When running as Web App in Firebase, data is stored in the FireStore. 

For a Desktop App, the location of the  is logged at the beginning of the task wherever you ran `npm run dev`. Data is also stored in a folder that is generated by the app, which should be found on the desktop.
```

**Review by RobertGemmaJr — CHANGES_REQUESTED** [review]
Couple spelling things here and there but otherwise looks great! Does this include the changes from #13? I don't think it's been merged in yet

File: docs/configuration.md [hunk 5]
```diff
@@ -5,27 +5,52 @@ title: Configuration
...
+
+- ![buildtime](https://img.shields.io/badge/-buildtime-blue): Build-time environment variable are configured before the application is built and cannot be changed at run-time. They are configured via `.env` files before building. Instead of using a single `.env` file, we prefer to defined separate files under the `env/` directory, and use a package called `dotenv-cli` before building`. 
+
+The `dotenv-cli` commes with the `dotenv` command that can be used to propery load the needed variables. This command does not run directly from the command line (terminal). Instead, to set build-time environment variables, we need to add a  script to `package.json` with the following format:
```
Comment (jashlu) [inline docs/configuration.md]: is this supposed to be `comes` instead of `commes`

**Review by gtdang — CHANGES_REQUESTED** [review]
Typo with "npm". My computer keeps autocorrecting npm to "nom" too! 😠 
Should we keep back-tics on the `npm` commands?

**Comment by mirestrepo** [comment]
> > Below we list the environment variables used by the app and indicate (via badges and text), whether they are mandatory, optional and needed during build (`npm build`) or run time (using the application executable or `npm run dev`).
> 
> I'm a bit confused by this sentence and the badges used in this guide. Perhaps splitting this sentence into two, 1) explain the optional/mandatory and 2) runtime/buildtime would be beneficial.
> 
> * Optional vs Mandatory - Should explain that the `optional` badge indicates optional and `default` + `default value` badge indicates mandatory?
> 
> _Suggested edit_ Below we list the environment variables used by the app and indicate their properties via badges and text. Optional variables are labeled with the optional badge and mandatory variables are labeled with the default false badge with their default value specified. The buildtime badge indicates the variable is used during the build/compilation phase (`npm build`) while the runtime badge indicates the variable is used during the executable run (`nom run dev`).

I love this suggestion!

File: docs/configuration.md (hunk 5 above)
Comment (mirestrepo) [reply docs/configuration.md]: Corrected, along with few other typos!

**Comment by mirestrepo** [comment]
I installed my vscode spilling plugin, sorry about all the typos!

**Review by gtdang — APPROVED** [review]

**Review by gtdang — APPROVED** [review]

File: docs/quick_start.md [hunk 6]
```diff
@@ -111,28 +117,15 @@ npm run dev
 
 This will launch an electron window with the task and inspector open. It will hot-reload whenever changes are made to the app.
 
-### 6. Run the task with preset environment variables
+### 6. Learn about configuring you task for diffent enviroments
```
Comment (hollandjg) [inline docs/quick_start.md]:
```suggestion
### 6. Learn about configuring your task for different environments
```

**Review by RobertGemmaJr — APPROVED** [review]
Looks great! The VS Code spellchecker is a life saver

**Pull request merged**